relay_test.go

```diff
@@ -0,0 +1,48 @@
+package tchannel_test
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	. "github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/raw"
+	"github.com/uber/tchannel-go/testutils"
+)
+
+func TestRelay(t *testing.T) {
+	relay, err := NewChannel("relay", &ChannelOptions{
+		RelayHosts: NewSimpleRelayHosts(map[string][]string{}),
+	})
+	require.NoError(t, err, "Failed to create a relay channel.")
+	require.NoError(t, relay.ListenAndServe("127.0.0.1:0"), "Relay failed to listen.")
+	defer relay.Close()
+
+	data := []byte("fake-body")
+
+	server := testutils.NewServer(t, testutils.NewOpts().SetServiceName("test"))
+	defer server.Close()
+	server.Register(raw.Wrap(newTestHandler(t)), "echo")
+	relay.RelayHosts().(*SimpleRelayHosts).Add("test", server.PeerInfo().HostPort)
+
+	client := testutils.NewClient(t, nil)
+	client.Peers().Add(relay.PeerInfo().HostPort)
+	defer client.Close()
+
+	var wg sync.WaitGroup
```

Comment:
this test seems weird -- it doesn't use `wg` and it defers code to run after the test function ends